From 77508db9c31234ddecb9bd8bc7b956d4d70b9f65 Mon Sep 17 00:00:00 2001
From: Mike Swanson <mikeonthecomputer@gmail.com>
Date: Mon, 3 Aug 2015 14:55:23 -0700
Subject: [PATCH 2/2] Makefile: clean up the install/uninstall targets

It referenced directories/files that only make sense for a game,
including installing the manpage in section 6 (games) rather than 1
and a non-exitent desktop icon and appdata.

I added the autotools-like prefix variable to configure the install
location, defaulting to /usr/local.
---
 Makefile | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index d86a888..15a6e87 100644
--- a/Makefile
+++ b/Makefile
@@ -18,19 +18,21 @@ all: git-weave.1
 clean:
 	rm -f *~ *.1 *.html *.rpm *.lsm MANIFEST
 
-install: git-weave.1 uninstall
-	install -m 0755 -d $(DESTDIR)/usr/bin
-	install -m 0755 -d $(DESTDIR)/usr/share/man/man6
-	install -m 0755 -d $(DESTDIR)//usr/share/applications/
-	install -m 0755 -d $(DESTDIR)/usr/share/pixmaps/
-	install -m 0755 -d $(DESTDIR)/usr/share/appdata
-	install -m 0755 git-weave $(DESTDIR)/usr/bin/
-	install -m 0644 git-weave.1 $(DESTDIR)/usr/share/man/man6/
-	install -m 0644 git-weave.xml $(DESTDIR)/usr/share/appdata/
+prefix?=/usr/local
+mandir?=share/man
+target=$(DESTDIR)$(prefix)
+
+install: git-weave.1
+	install -d "$(target)/bin"
+	install -d "$(target)/$(mandir)/man1"
+	install -m 755 git-weave "$(target)/bin"
+	install -m 644 git-weave.1 "$(target)/$(mandir)/man1"
 
 uninstall:
-	rm -f /usr/bin/git-weave /usr/share/man/man6/git-weave.1
-	rm -f /usr/share/pixmaps/git-weave-logo.png
+	rm "$(target)/$(mandir)/man1/git-weave.1"
+	rm "$(target)/bin/git-weave"
+	-rmdir -p "$(target)/$(mandir)/man1"
+	-rmdir -p "$(target)/bin"
 
 PYLINTOPTS = --rcfile=/dev/null --reports=n \
 	--msg-template="{path}:{line}: [{msg_id}({symbol}), {obj}] {msg}" \
-- 
2.5.0

